File: server/models/sub_categoryModel.js
const mongoose = require("mongoose");

const subCategorySchema = new mongoose.Schema({
  categoryID: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "Category",
    required: true,
  },
  sub_category_type: {
    type: String,
    enum: ["components", "accessories"],
    required: true,
  },
  sub_category_list: [{ name: String, image: String }],
});

const SubCategory = mongoose.model("SubCategory", subCategorySchema);

module.exports = SubCategory;

File: admin_panel/lib/feature/create_product/model/sub_category_model.dart
class SubCategory {
  final String id;
  final String name;
  final String image;
  final String childType;

  SubCategory({
    required this.id,
    required this.name,
    required this.image,
    required this.childType,
  });

  factory SubCategory.fromJson(Map<String, dynamic> json) {
    return SubCategory(
      id: json['_id'] ?? '',
      name: json['name'] ?? 'Chưa có tên',
      image: json['image'] ?? '',
      childType: json['child_type'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'image': image,
      'child_type': childType,
    };
  }
}

